Plugins/NativeCamera/Source/Android/CameraWrappers.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <android/api-level.h>
+#include <arcana/macros.h>
+#include <dlfcn.h>
+#include <map>
+
+// The camera API's in the NDK are only supported in API24 and higher. We
+// use dlopen and dlsym to link the camera lib at runtime only if the current
+// device supports it. In order to avoid redefining the type information the
+// below macro's forcefully import the camera header files as if the min supported
+// SDK was set to 24.
+#define __ORIG_ANDROID_API__ __ANDROID_API__
+#undef __ANDROID_API__
+#define __ANDROID_API__ 24
+namespace API24 {
+#include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraCaptureSession.h>
+    #include <camera/NdkCameraDevice.h>
+    #include <camera/NdkCameraError.h>
+    #include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraMetadata.h>
+    #include <camera/NdkCameraMetadataTags.h>
+    #include <camera/NdkCameraWindowType.h>
+    #include <camera/NdkCaptureRequest.h>
+}
+#undef __ANDROID_API__
+#define __ANDROID_API__ __ORIG_ANDROID_API__
+#undef __ORIG_ANDROID_API__
+
+// Helper macro to make calling camera NDK api's more type safe
+#define GET_CAMERA_FUNCTION(function) reinterpret_cast<decltype(&API24::function)>(GetCameraDynamicFunction(#function))
+
+static const int API_LEVEL{ android_get_device_api_level() };
+
+// Load the NDK camera lib dynamically. When running on OS 6.0 and below this will return nullptr,
+// on OS 7.0 and up this should return a pointer to access the library using dlsym. It is technically fine
+// to call dlopen when the API_LEVEL is below 24, but it's wasted effort on those devices.
+static void* libCamera2NDK{ API_LEVEL >= 24 ? dlopen("libcamera2ndk.so", RTLD_NOW | RTLD_LOCAL): nullptr };
```

Comment:
What happens if this library has already been implicitly loaded because the consuming application is API level 24+ and it explicitly uses the Camera2 NDK APIs itself? Does it return the handle to the already loaded library? Or does it return null? Or throw?

Reply:
I tested this out by re-enabling the CMAKE file that causes the consumer to link against the libcamera2ndk and then ran it in the Playground app (with the min API set to 24). Everything still worked correctly when running the app on device with a camera feed being rendered in the scene.

I'm not sure if the pointer we get from `dlopen` is the same one that would be used internally for the statically linked library, but it does at least work as intended still.